linera-views/src/reentrant_collection_view.rs

```diff
@@ -134,7 +134,17 @@ where
             return true;
         }
         let updates = self.updates.lock().await;
-        !updates.is_empty()
+        for update in updates.values() {
+            match update {
+                Update::Removed => return true,
+                Update::Set(view) => {
+                    if view.read().await.has_pending_changes().await {
+                        return true;
+                    }
+                }
+            }
+        }
+        false
```

Comment:
Is this definition of `has_pending_changes` consistent with `clear` and `hash`?

Reply:
Oops, yeah this could return `false` if a new entry is inserted into the collection. I changed the fix and added a unit test for that scenario.